src/main/java/org/inferred/freebuilder/processor/DefaultPropertyFactory.java

```diff
@@ -169,18 +169,56 @@ public void addFinalFieldAssignment(SourceBuilder code, String finalField, Strin
 
     @Override
     public void addMergeFromValue(Block code, String value) {
-      code.addLine("%s(%s.%s());", setter(property), value, property.getGetterName());
+      Excerpt defaults = Declarations.freshBuilder(code, metadata).orNull();
+      if (defaults != null) {
+        code.add("if (");
+        if (!hasDefault) {
+          code.add("%s._unsetProperties.contains(%s.%s) || ",
```

Comment:
I know the tests are passing, but I'm confused about why the `_unsetProperties` check is done on the defaults and not on the `template/value` that is being merged in. 

I would expect the logic to be `if this property is set (aka, different from its default value) on the builder / value I'm merging into this builder and the value for the property is different from mine, set it`